src/main/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -107,11 +110,27 @@ public void schedule(ScheduleRequest scheduleRequest) {
 		catch(ParseException pe) {
 			throw new IllegalArgumentException("Cron Expression is invalid: " + pe.getMessage());
 		}
-
-		retryTemplate().execute(e -> {
-			scheduleTask(appName, scheduleName, cronExpression, command);
-			return null;
-		});
+		retryTemplate().execute(new RetryCallback<Void, RuntimeException>() {
```

Comment:
Not part of this PR, but there isn't a need to recreate the retryTemplate over and over again, a singleton bean instance  would work.  Later can open up the configuration of the template itself, number of retries etc.